compiler/src/dotty/tools/dotc/transform/Splicer.scala

```diff
@@ -264,9 +264,12 @@ object Splicer {
       }
 
       val name = getDirectName(fn.info.finalResultType, fn.name.asTermName)
-      val method = getMethod(clazz, name, paramsSig(fn))
-
-      (args: List[Object]) => stopIfRuntimeException(method.invoke(inst, args: _*))
+      if (name == nme.asInstanceOfPM)
```

Comment:
This check should be done before we call `interpretedStaticMethodCall`. I should be around https://github.com/lampepfl/dotty/pull/6844/files#diff-43be0f2f8a9625bb70763569c65a6fdcR190